packages/lit-ssr/src/lib/render-lit-html.ts

```diff
@@ -53,30 +53,58 @@ import {
   isElement,
 } from './util/parse5-utils.js';
 
-import {isRenderLightDirective} from 'lit-html/directives/render-light.js';
-import {LitElement} from 'lit-element';
+import {isRenderLightDirective} from 'lit/directives/render-light.js';
+import {LitElement} from 'lit';
 import {LitElementRenderer} from './lit-element-renderer.js';
 import {reflectedAttributeName} from './reflected-attributes.js';
-import {Directive} from 'lit-html/directive';
 
 declare module 'parse5' {
   interface DefaultTreeElement {
     isDefinedCustomElement?: boolean;
   }
 }
 
-patchDirectiveResolve(
-  Directive.prototype,
-  function (this: Directive, values: unknown[]) {
-    const {__part, __attributeIndex} = this;
-    return resolveDirective(
-      __part,
-      this.render(...values),
-      this,
-      __attributeIndex
+/**
+ * Looks for values of type `DirectiveResult` and patches a `Directive` class's
+ * implementation of `_$resolve` to call `render` rather than `update`, per the
+ * SSR contract
+ */
+const patchIfDirective = (value: unknown) => {
+  const directiveCtor = (value as DirectiveResult)?._$litDirective$;
+  if (directiveCtor !== undefined) {
+    patchDirectiveResolve(
```

Comment:
It seems safer to me to subclass the directive to override resolve

Reply:
Agree, if it's chill.

Reply:
Done, changed the helper exported in `private-ssr-support` to be a mixin that overrides resolve to render, and then SSR uses a cache to avoid subclassing every DirectiveResult's directive anew.

I also relocated where `resolveDirective` is called on the return value of a directive (to support nested directives), since it was unsafely dereferencing possibly unstable class fields from `Part`. This has the benefit of also making the mixin override simpler.